extension/encoding/awslogsencodingextension/internal/unmarshaler/cloudtraillogs/cloudtraillogs.go

```diff
@@ -0,0 +1,281 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package cloudtraillogs
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+// CloudTrailLogsUnmarshaler is an unmarshaler for CloudTrail logs.
+type CloudTrailLogsUnmarshaler struct {
+	buildInfo component.BuildInfo
+}
+
+var _ plog.Unmarshaler = (*CloudTrailLogsUnmarshaler)(nil)
+
+// CloudTrailRecord represents a CloudTrail log record
+type CloudTrailRecord struct {
+	EventVersion                 string         `json:"eventVersion"`
+	EventTime                    string         `json:"eventTime"`
+	EventSource                  string         `json:"eventSource"`
+	EventName                    string         `json:"eventName"`
+	AwsRegion                    string         `json:"awsRegion"`
+	SourceIPAddress              string         `json:"sourceIPAddress"`
+	UserAgent                    string         `json:"userAgent"`
+	RequestID                    string         `json:"requestID"`
+	EventID                      string         `json:"eventID"`
+	EventType                    string         `json:"eventType"`
+	EventCategory                string         `json:"eventCategory"`
+	RecipientAccountID           string         `json:"recipientAccountId"`
+	UserIdentity                 map[string]any `json:"userIdentity"`
+	ResponseElements             map[string]any `json:"responseElements"`
+	RequestParameters            map[string]any `json:"requestParameters"`
+	Resources                    []any          `json:"resources"`
+	ReadOnly                     bool           `json:"readOnly"`
+	ManagementEvent              bool           `json:"managementEvent"`
+	AdditionalEventData          map[string]any `json:"additionalEventData"`
+	TLSDetails                   map[string]any `json:"tlsDetails"`
+	SessionCredentialFromConsole string         `json:"sessionCredentialFromConsole"`
+}
+
+const (
+	cloudProvider = "aws"
+	scopeName     = "cloudtrail.processor"
+)
+
+func NewCloudTrailLogsUnmarshaler(buildInfo component.BuildInfo) plog.Unmarshaler {
+	return &CloudTrailLogsUnmarshaler{
+		buildInfo: buildInfo,
+	}
+}
+
+func (u *CloudTrailLogsUnmarshaler) Unmarshal(buf []byte) (plog.Logs, error) {
+	return u.UnmarshalLogs(buf)
+}
+
+func (u *CloudTrailLogsUnmarshaler) UnmarshalLogs(buf []byte) (plog.Logs, error) {
```

Comment:
By the way, the CloudTrail log files are also compressed (see [documentation](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-log-file-examples.html)). So this unmarshaler would need the same `sync.Pool` of gzip readers the others are using too.

I think this part could be removed from the unmarshalers in the future so we could just use that code one time, but for now we have to keep repeating it.